### PR TITLE
chore(RHINENG-11401): Refactor RBAC

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-11401](https://issues.redhat.com/browse/RHINENG-11401).
It refactors the RBAC implementation, extracts logical blocks of the code into their own functions, and simplifies its usage within export_service.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: lib/middleware.py [hunk 1]
```diff
@@ -75,96 +70,139 @@ def get_rbac_permissions(app, request_header=None):
...
+    rbac_request_headers: dict,
+    permission_base: str = "inventory",
+) -> Tuple[str, dict]:
+    # Returns a 2-item tuple:
+    # 1. Whether or not the request should be allowed at all
+    # 2. The filter that should be applied to the data (if allowed)
+    if inventory_config().bypass_rbac:
+        return True, None
+
+    current_identity = get_current_identity()
```
Comment (jpramos123) [inline lib/middleware.py:137]: I'm afraid it will fail in the export-service. As we are not receiving an API call, the application does not create a new context, so the `connexion.context.context["user"]` attribute will not exist. The app runs once, in a infinite loop, and it leaves the normal API pipeline that Flask creates when receiving API calls

That is the reason we were using `current_identity = create_mock_identity_with_org_id(org_id)` to create the identity, because we needed to mock it every time we received an message from Kafka.

I would suggest validation if the `identity` exists, if not get it using `get_current_identity()`

```suggestion
   if not identity:
       current_identity = get_current_identity()
```

**Review by jpramos123 — CHANGES_REQUESTED** [review]
Just one comment

File: lib/middleware.py (hunk 1 above)
Comment (kruai) [reply lib/middleware.py:137]: My mistake; this was leftover from the original implementation. This line doesn't need to be here. When coming from `rbac()`, the identity is already acquired from the connexion context. When coming from the export service, it's already set to the mock identity.

**Review by jpramos123 — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]
